devnet-sdk/devstack/sysgo/l2_cl.go

```diff
@@ -123,11 +131,46 @@ func WithL2CLNode(l2CLID stack.L2CLNodeID, isSequencer bool, l1CLID stack.L1CLNo
 
 		jwtPath, jwtSecret := orch.writeDefaultJWT()
 
-		var p2pSigner *p2p.PreparedSigner
-		if isSequencer {
-			p2pKey, err := orch.keys.Secret(devkeys.SequencerP2PRole.Key(l2CLID.ChainID.ToBig()))
-			require.NoError(err, "need p2p key for sequencer")
-			p2pSigner = &p2p.PreparedSigner{Signer: opsigner.NewLocalSigner(p2pKey)}
+		logger := o.P().Logger().New("service", "op-node", "id", l2CLID)
+
+		var p2pSignerSetup p2p.SignerSetup
+		var p2pConfig *p2p.Config
+		// code block for P2P setup
+		{
+			// make a dummy flagset since p2p config initialization helpers only input cli context
+			fs := flag.NewFlagSet("", flag.ContinueOnError)
+			// use default flags
+			for _, f := range opNodeFlags.P2PFlags(opNodeFlags.EnvVarPrefix) {
+				require.NoError(f.Apply(fs))
+			}
+			// mandatory P2P flags
+			require.NoError(fs.Set(opNodeFlags.AdvertiseIPName, "127.0.0.1"))
+			require.NoError(fs.Set(opNodeFlags.AdvertiseTCPPortName, "0"))
+			require.NoError(fs.Set(opNodeFlags.AdvertiseUDPPortName, "0"))
+			require.NoError(fs.Set(opNodeFlags.ListenIPName, "127.0.0.1"))
+			require.NoError(fs.Set(opNodeFlags.ListenTCPPortName, "0"))
+			require.NoError(fs.Set(opNodeFlags.ListenUDPPortName, "0"))
+			// avoid resource unavailable error by using memorydb
+			require.NoError(fs.Set(opNodeFlags.DiscoveryPathName, "memory"))
+			require.NoError(fs.Set(opNodeFlags.PeerstorePathName, "memory"))
+			// For peer ID
+			networkPrivKey, err := crypto.GenerateKey()
+			require.NoError(err)
+			networkPrivKeyHex := hex.EncodeToString(crypto.FromECDSA(networkPrivKey))
+			_ = fs.Set(opNodeFlags.P2PPrivRawName, networkPrivKeyHex)
+
+			cliCtx := cli.NewContext(&cli.App{}, fs, nil)
```

Comment:
I think there is another way of doing this without temporary CLI context, by using the prepared-p2p setup. I think that is used in op-e2e already, when p2p is enabled (although not used much there, and wrapping a mock p2p network with in-process communication pipes instead of network sockets).

Reply:
By doing this, the sysgo behavior will differ from how "actual" op-node behaves. I believe testing the code with the actual used p2p stack in op-node is also beneficial. So ideally, we may have a pluggable p2p which can be prepared-p2p or using the cliCtx as like the current implementation. 

My proposal is to make a separate PR for using the prepared-p2p setup. Also need some exploration how the the prepared p2p can be embedded to the current devstack / sysgo architecture.

Reply:
@protolambda tracking this at https://github.com/ethereum-optimism/optimism/issues/15479